NeoML/include/NeoML/Dnn/DnnDistributed.h

```diff
@@ -102,4 +105,38 @@ class NEOML_API CDistributedTraining {
 	friend class CLoraSerializer;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+
+// Single process, multiple threads distributed inference on CPU
+class NEOML_API CDistributedInference {
+public:
+	// Creates `count` cpu models
+	// If `count` is 0 or less, then the models number equal to the number of available CPU cores
+	CDistributedInference( CDnn& dnn, int count );
+	CDistributedInference( IMathEngine& mathEngine, CArchive& archive, int count, int seed = 42 );
```

Comment:
I've found no checks that mathEngine is a cpu engine (except comments). Are there any reasonable scenarios when we would like to run this s*** on GPU? Or I just missed some checks?